### PR TITLE
Add functionality for "auth and commit", or "auth and capture": Add t…

### PR DESCRIPTION
**Pull request by Swie**
…ransactionResult method for checking out whether or not a Form payment OR debit transaction with commit=false, succeeded without actually committing it. Add commit boolean and committedAmount to commit, transaction status and order search responses. Committed amount is especially useful if an amount smaller than the authorized payment was committed.

### CODE REVIEW

File: src/main/java/io/paymenthighway/json/JsonParser.java [hunk 1]
```diff
@@ -53,6 +53,11 @@ public CommitTransactionResponse mapCommitTransactionResponse(String json) {
   }
 
   @Deprecated
+  public TransactionResultResponse mapTransactionResultResponse(String json) {
```
Comment (margussipria) [inline src/main/java/io/paymenthighway/json/JsonParser.java]: Why was this added? this isn't even used?

Comment (Swie) [inline src/main/java/io/paymenthighway/json/JsonParser.java]: Removed
